### ISSUE TITLE
Add option to force storage type to be explicit

### ISSUE THREAD

**Issue by linusnorton** [protofire/solhint#28]
It would be nice if I could specify an option that enforced explicit declaration of storage type.

For example

```
function something(string foo) public { ... }
```

should specify a storage type of `memory` or `storage` for the parameter `foo`.

**Comment by mgarciap** [comment]
@pablofullana, @mariano-aguero, @fvictorio, @idrabenia ?

**Comment by mariano-aguero** [comment]
Hi @linusnorton , how are you ? Thanks for contributing to solhint, we really  appreciate.

I'm worried about this possible rule, I was analyzing the implementation and I found many situations that make some noise:
  - We cannot distinguish between structs and contracts
  - We will have many false positives
  - I think a rule that does not detect 100% of the cases makes no sense (This is a personal opinion as developer)

I would like to know what do you think about my concerns. 

Regards!

**Comment by linusnorton** [comment]
I agree, it should work for everything or not at all
